### PR TITLE
chore: CE-556 HWCR drug administration information updates

### PR DESCRIPTION
**Pull request by joshgamache**
Addresses [CE-556](https://env-ceds.atlassian.net/browse/CE-556). Changes to some fields and available options when entering Drug information for animals. 

**Note**: This PR is paired with an [equivalent PR in the non-CM repo](https://github.com/bcgov/nr-compliance-enforcement-cm/pull/101).

## Changes 🚧

- Added _Oral/nasal_ to options in 'Injection method'.
- Replaced an option in 'Fate of remaining...' with _Returned to vet_. 
- Removed _Adverse reactions_, _Discard method_, and _Amount discarded_ fields.
- Added _Additional comments_ text field.
- (In CM DB) Migrated removed fields' data into _Additional comments_ data. 

## How Has This Been Tested? 🔬

- [x] Manual testing, matching  from issue report.
- [x] Dummy data in DB for migration.

### To test 🔬

0. Ensure both this and `...cm` repo are on this issue's branch.
1. Run a local instance of the application, log in. 
2. Navigate to a HWC complaint and **Add animal**, then **Add drug**.
   - You should no longer see the removed fields. You should now see an 'Additional comments' open text field. 
   - Under 'Injection Method', you should now see a fourth option _Oral/nasal_
   - Under 'Fate of remaining drug in vial', you should not see _Reverse distribution_ but instead see _Returned to vet_
3. Fill out the fields, and save the 'Animal outcome'. 

### Checklist 📃

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

[CE-556]: https://env-ceds.atlassian.net/browse/CE-556?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-compliance-enforcement-cm-101-backend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement-cm/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement-cm/actions/workflows/merge.yml)

### CODE REVIEW

File: migrations/sql/V1.30.0__CE-556.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+UPDATE case_management.drug_remaining_outcome_code
+  SET short_description='Returned to vet', long_description='Returned to vet', update_user_id='FLYWAY', update_utc_timestamp=now()
+  WHERE drug_remaining_outcome_code='RDIS';
+
+--
+-- CREATE TABLE drug_additional_comments
+--
+
+ALTER TABLE case_management.drug_administered
+  ADD additional_comments_text varchar(1000) NULL;
```
Comment (afwilcox) [inline migrations/sql/V1.30.0__CE-556.sql:19]: Our current standard for large blocks of text like this (Notes, Complaint Details, Suspect Witness details) is to use text type.

File: backend/src/case_file/case_file_inputs.graphql [hunk 2]
```diff
@@ -200,6 +200,20 @@ input DrugInput {
...
+  amountUsed: String!
+  injectionMethod: String!
+
+  remainingUse: String
+  additionalComments: String
+
+  order: Int
+}
+
 input WildlifeActionInput {
```
Comment (afwilcox) [inline backend/src/case_file/case_file_inputs.graphql:217]: As we are currently the only consumer of the API, do we need to version the input?

**Review by afwilcox — CHANGES_REQUESTED** [review]

File: backend/src/case_file/case_file_inputs.graphql (hunk 2 above)
Comment (joshgamache) [reply backend/src/case_file/case_file_inputs.graphql]: Corrected this with ea9d7b3.

File: migrations/sql/V1.30.0__CE-556.sql (hunk 1 above)
Comment (joshgamache) [reply migrations/sql/V1.30.0__CE-556.sql]: Revised the column with cd87468.

File: migrations/sql/V1.30.0__CE-556.sql
[the remark this answers is not in the document]
Comment (joshgamache) [reply migrations/sql/V1.30.0__CE-556.sql:1]: Thanks for the info, fixed the file naming with cd87468.

**Review by afwilcox — APPROVED** [review]

**Pull request merged**